pkg/scrape/target.go

```diff
@@ -100,6 +102,15 @@ func (t *Target) offset(interval time.Duration) time.Duration {
 	return time.Duration(next)
 }
 
+func (t *Target) SpyName() string {
+	for _, l := range t.discoveredLabels {
+		if l.Name == model.SpyNameLabel && l.Value != "" {
+			return l.Value
+		}
+	}
+	return spyName
```

Comment:
Maybe we should signal somehow that it's defaulting to a deprecated spyName?

Reply:
Yeah, it's definitely is a good idea. I peeked and it seems it's a bigger change that it might seem as we need to change `SpyName()` return type to something like `(string, error)` but it doesn't supported along the code yet. Let me create separate issue for that

Reply:
https://github.com/pyroscope-io/pyroscope/issues/1066